vite/msdfGenerator.js

```diff
@@ -158,6 +163,44 @@ const copyDir = (src, dest) => {
     const srcPath = path.join(src, entry.name)
     const destPath = path.join(dest, entry.name)
 
+    // Skip copying charset.txt file into dist directory
+    if (srcPath.includes('.charset.txt')) continue
+
     entry.isDirectory() ? copyDir(srcPath, destPath) : fs.copyFileSync(srcPath, destPath) // Copy files
   }
 }
+
+const createCharsetFile = (charsetDir, ConfigDir, fontName) => {
+  const charsetFilePath = path.join(charsetDir, `${fontName}.charset.txt`)
+
+  // Check charset file already exists, if exists, return file path
+  if (fs.existsSync(charsetFilePath)) {
+    return charsetFilePath
+  }
+
+  const configFilePath = path.join(ConfigDir, `${fontName}.config.json`)
+
+  // Check if font config.json is exists
+  if (fs.existsSync(configFilePath)) {
+    if (!fs.existsSync(charsetDir)) {
+      fs.mkdirSync(charsetDir, { recursive: true })
+    }
+    const configData = JSON.parse(fs.readFileSync(configFilePath, 'utf8'))
+    let charsetContent = configData.charset
+    const presets = configData.preset
+    for (let i = 0; i < presets.length; i++) {
+      charsetContent = charsetContent + presets[i]
```

Comment:
this would just add the preset name to the charset, instead of appending the characters associated with the preset, right?

Reply:
No, It will append all characters in preset array to charset

Reply:
Wait, actually locally I have different logic let me check again.

Reply:
Adding presets defintion to charset logic moved to msdf-generator so this comment can be out of blits vite plugin scope.  Resolving this conversation.